### ISSUE TITLE
restrict package by version

### ISSUE THREAD

**Issue by rveltz** [JuliaLang/Pkg.jl#3925]
Hi,

I lost a manifest and I know when I ran some code. Say November 2022. I want to install packages at the latest version given the time uper bound  November 2022.

Would it be possible to do so?

Best regards

**Comment by KristofferC** [comment]
You can clone the general registry and check it out at a commit at that time.

**Comment by rveltz** [comment]
Yes, I was told the same solution on discourse. I merely open this issue to see if anybody would be interested in a solution within `Pkg`

**Comment by cstjean** [comment]
In details (thank you @aplavin, @giordano):

1. In the `pkg>` prompt, type `registry add git@github.com:JuliaRegistries/General`. This creates a `.julia/registries/General` folder, which is a `git clone` of the general registry
2. Go to `.julia/registries` and delete/rename `General.toml`, so that the default up-to-date registry is skipped. Pkg will fall back to the `/General` folder's registry
3. Go to `/General`, type `git rev-list -1 --before="2026-01-22" master` (with the desired date) to get the commit from that date.
4. Check out that commit
5. `]up` your packages to get the packages from that date

Then `git bisect` etc. can be used inside `/General` to find the culprit.
